### PR TITLE
The RPC Framework crust package

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
@soltanmm please begin review - the "DO NOT MERGE" is because I haven't the time for one more test execution before supper. :-P

Also this is blocked on at least some of today's earlier pull requests, including at least one of the still-open ones... I think.

### CODE REVIEW

**Comment by nathanielmanistaatgoogle** [comment]
Amended and reviewable, though I think #3047 is a prerequisite.

**Comment by nathanielmanistaatgoogle** [comment]
Please proceed with review and merge.

**Comment by nathanielmanistaatgoogle** [comment]
Python-green.

File: src/python/grpcio/grpc/framework/crust/_control.py [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+    rpc_abortion_callback, rendezvous_set_outcome):
+  def operation_termination_callback(operation_outcome):
+    termination = rendezvous_set_outcome(operation_outcome)
+    if termination.abortion is not None:
+      rpc_abortion_callback(termination.abortion)
+  return operation_termination_callback
+
+
+class Rendezvous(base.Operator, future.Future, stream.Consumer, face.Call):
+  """A rendez-vous for the threads of an operation.
```
Comment (soltanmm) [inline src/python/grpcio/grpc/framework/crust/_control.py]: Just out of curiosity, why's there a dash here?

File: src/python/grpcio_test/grpc_test/framework/interfaces/face/_3069_test_constant.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""A test constant working around issue 3069."""
+
+# test_constants is referenced from specification in this module.
+from grpc_test.framework.common import test_constants  # pylint: disable=unused-import
+
+# TODO(issue 3069): Replace uses of this constant with
+# test_constants.SHORT_TIMEOUT.
+REALLY_SHORT_TIMEOUT = 0.1
```
Comment (soltanmm) [inline src/python/grpcio_test/grpc_test/framework/interfaces/face/_3069_test_constant.py]: How is this related to #3069?

File: src/python/grpcio/grpc/framework/crust/_control.py (hunk 1 above)
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/framework/crust/_control.py]: Apparently that's the spelling that I used when I wrote the comment... one year ago. I'm not attached to it and can drop the dash if you think it helps the reader.

File: src/python/grpcio_test/grpc_test/framework/interfaces/face/_3069_test_constant.py (hunk 2 above)
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio_test/grpc_test/framework/interfaces/face/_3069_test_constant.py]: When I try to run this test with `test_constants.SHORT_TIMEOUT` in this place, test execution times out. Maybe there's yet another ceiling in addition to the ones of which we're already aware?

Comment (soltanmm) [inline src/python/grpcio_test/grpc_test/framework/interfaces/face/_3069_test_constant.py]: That's... really really weird.
